### PR TITLE
[Osquery] Extend kibana_system role with an access to osquery_manager…

### PR DESCRIPTION
**Pull request by patrykkopycinski**
In order to support transforms we need to extend the current user with a privilege to create `responses` index https://github.com/elastic/elasticsearch/pull/84941#issuecomment-1081974019

Follow up for https://github.com/elastic/elasticsearch/pull/84279

```
The PR adds proper permissions for the kibana_system user to two indices specific to Osquery actions.

The two indices are .logs-osquery_manager.actions-* and .logs-osquery_manager.action.responses-* .

The kibana_system user will create, write, and read from the .logs-osquery_manager.actions-* index to log new actions and visualize them for users.

The kibana_system user will read from .logs-osquery_manager.action.responses-*` to visualize responses from the Osquery for the user.

This change is needed as Osquery plugins currently are using raw Fleet actions, but in order to provide a better user experience, we need to store additional metadata about the actions that are taken. We need to ensure kibana_system can properly interact with these indices.

Adds access to indices added here https://github.com/elastic/integrations/pull/2563
Follows https://github.com/elastic/elasticsearch/pull/84279
```

### CODE REVIEW

**Comment by ywangd** [comment]
ping @elastic/kibana-security

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java
Comment (azasypkin) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java:773]: __nit:__ would make sense to update the comment as well to reflect new privileges and what are they used for.

**Review by azasypkin — APPROVED** [review]
LGTM from the Kibana Security perspective.

**Comment by ywangd** [comment]
@patrykkopycinski Next time please also wait for a "LGTM" from the @elastic/es-security team before merging. Thanks!

**Pull request merged**
